tfx/extensions/experimental/kfp_compatibility/proto/kfp_component_spec.proto

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+syntax = "proto3";
+// protolint:disable FIELD_NAMES_LOWER_SNAKE_CASE
+// protolint:disable PACKAGE_NAME_LOWER_CASE
+// protolint:disable REPEATED_FIELD_NAMES_PLURALIZED
+
+package tfx.extensions.experimental.kfp_compatibility;
```

Comment:
JFYI: Here is the proto variant of the KFP's ComponentSpec: https://github.com/kubeflow/pipelines/blob/db0af86e53ac050774b9331cd89a74dcdf04d1c4/sdk/python/kfp/components/structures/components.proto#L65

There are some differences since the real schema can only be expressed as JsonSchma, but not proto.

The JsonSchema is here: https://github.com/kubeflow/pipelines/blob/db0af86e53ac050774b9331cd89a74dcdf04d1c4/sdk/python/kfp/components/structures/components.json_schema.json#L190